doc/source/timeseries.rst

```diff
@@ -265,17 +265,34 @@ Typical epoch stored units
    pd.to_datetime([1349720105100, 1349720105200, 1349720105300,
                    1349720105400, 1349720105500 ], unit='ms')
 
-These *work*, but the results may be unexpected.
+.. note::
 
-.. ipython:: python
+   Epoch times will be rounded to the nearest nanosecond.
+
+.. warning::
+
+   Conversion of float epoch times can lead to inaccurate and unexpected results.
+   Python floats are unbounded precision, with about 15 significant digits (depends on
+   operating system). Rounding in conversion from to high precision ``Timestamp`` is
+   unavoidable. The only way to have exact precision is to use a fixed-width exact type
+   (e.g. an int64).
 
-   pd.to_datetime([1])
+   .. ipython:: python
 
-   pd.to_datetime([1, 3.14], unit='s')
+      1490195805.433502912
+      pd.to_datetime([1490195805.433, 1490195805.433502912], unit='s')
+      1490195805433502912
+      pd.to_datetime(1490195805433502912, unit='ns')
 
 .. note::
 
-   Epoch times will be rounded to the nearest nanosecond.
+   These *work*, but the results may be unexpected.
```

Comment:
this part is now hanging, so add something like. numeric input is in referece to the ``epoch`` time (actually may want to move this mini-example (just this last part here) to the 'origin' section.)